spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -411,6 +475,43 @@ def name_id_version(format_urn)
       end
     end
 
+    context 'service provider has multiple certs' do
+      let(:service_provider) do
+        create(:service_provider,
+               cert: nil, # override singular cert
+               certs: ['saml_test_sp2', 'saml_test_sp'],
+               active: true)
+      end
+
+      let(:first_cert_settings) do
+        saml_settings.tap do |settings|
+          settings.issuer = service_provider.issuer
+        end
+      end
+
+      let(:second_cert_settings) do
+        saml_settings.tap do |settings|
+          settings.issuer = service_provider.issuer
+          settings.certificate = File.read(Rails.root.join('certs', 'sp', 'saml_test_sp2.crt'))
+          settings.private_key = OpenSSL::PKey::RSA.new(
+            File.read(Rails.root + 'keys/saml_test_sp2.key'),
+          ).to_pem
+        end
+      end
+
+      it 'encrypts the response to the right key' do
+        user = create(:user, :signed_up)
+        generate_saml_response(user, second_cert_settings)
+
+        expect(response).to_not be_redirect
+
+        expect { xmldoc.saml_response(first_cert_settings) }.to raise_error
```

Comment:
this is the assertion I was relying on the most to make sure that I wasn't magically defaulting to a wrong cert, ex if we change `matching_cert || service_provider.ssl_certs.last` this will fail